scripts/hail_batch/hgdp1kg_tobwgs_pca/hgdp_1kg_tob_wgs_pca.py

```diff
@@ -0,0 +1,77 @@
+"""Generates loadings, scores, and eigenvalues for the HGDP,1KG, and tob-wgs dataset"""
+
+import click
+import pandas as pd
+import hail as hl
+from hail.experimental import lgt_to_gt
+
+GNOMAD_HGDP_1KG_MT = (
+    'gs://gcp-public-data--gnomad/release/3.1/mt/genomes/'
+    'gnomad.genomes.v3.1.hgdp_1kg_subset_dense.mt'
+)
+
+TOB_WGS = 'gs://cpg-tob-wgs-main/joint_vcf/v1/raw/genomes.mt'
+
+GNOMAD_LIFTOVER_LOADINGS = 'gs://cpg-reference/gnomad/gnomad_loadings_90k_liftover.ht'
+
+
+@click.command()
+@click.option('--output', help='GCS output path', required=True)
+def query(output):  # pylint: disable=too-many-locals
+    """Query script entry point."""
+
+    hl.init(default_reference='GRCh38')
+
+    mt_hgdp_1kg = hl.read_matrix_table(GNOMAD_HGDP_1KG_MT)
+    mt_tob_wgs = hl.read_matrix_table(TOB_WGS)
+    # make keys the same between datasets
+    mt_tob_wgs = mt_tob_wgs.key_rows_by('locus', 'alleles')
+    # add gnomad liftover loading data to both matrices
+    ht_gnomad_loadings_liftover = hl.read_table(GNOMAD_LIFTOVER_LOADINGS)
+    ht_gnomad_loadings_liftover = ht_gnomad_loadings_liftover.key_by('locus', 'alleles')
+    join_rows_hgdp1kg_liftover = mt_hgdp_1kg.semi_join_rows(ht_gnomad_loadings_liftover)
+    join_rows_tobwgs_liftover = mt_tob_wgs.semi_join_rows(ht_gnomad_loadings_liftover)
+
+    # Join datasets by merging columns
+    # Entries and columns must be identical
+    change_lgt_to_gt = join_rows_tobwgs_liftover.annotate_entries(
+        GT=lgt_to_gt(join_rows_tobwgs_liftover.LGT, join_rows_tobwgs_liftover.LA)
+    )
+    select_entries_hgdp1kg = join_rows_hgdp1kg_liftover.select_entries(
+        join_rows_hgdp1kg_liftover.GT
+    )
+    select_entries_tobwgs = change_lgt_to_gt.select_entries(change_lgt_to_gt.GT)
+
+    # Only keep columns we need
+    select_columns_hgdp1kg = select_entries_hgdp1kg.select_cols(
+        select_entries_hgdp1kg.sample_qc
+    )
+    hgdp1kg_filtered = select_columns_hgdp1kg.drop('sample_qc')
+    tobwgs_filtered = select_entries_tobwgs
```

Comment:
Instead of creating all these new names, I think it's more common in Hail scripts to simple reassign to the previous variable names, unless you need them in a later expression.

Reply:
that makes sense. I think instead of leaving (a lot of) comments (which I'm normally used to doing), I created sample names that reflected what I was doing. But I will make sure to follow that for future scripts :)